src/applications/app.helpers.js

```diff
@@ -74,3 +74,111 @@ export function isParcel(appOrParcel) {
 export function objectType(appOrParcel) {
   return isParcel(appOrParcel) ? "parcel" : "application";
 }
+
+export function validateArgumentsTypeAPI(
+  appName,
+  applicationOrLoadingFn,
+  activityFn,
+  customProps
+) {
+  if (typeof appName !== "string" || appName.length === 0)
+    throw Error(
+      formatErrorMessage(
+        20,
+        __DEV__ &&
+          `The first argument to registerApplication must be a non-empty string 'appName'`
+      )
+    );
+
+  if (!applicationOrLoadingFn)
+    throw Error(
+      formatErrorMessage(
+        23,
+        __DEV__ && "The application or loading function is required"
+      )
+    );
+
+  if (typeof activityFn !== "function")
+    throw Error(
+      formatErrorMessage(
+        24,
+        __DEV__ && `The activityFunction argument must be a function`
+      )
+    );
+
+  if (
+    !!customProps &&
+    (typeof customProps !== "object" || Array.isArray(customProps))
+  )
+    throw Error(
+      formatErrorMessage(
+        22,
+        __DEV__ && "optional customProps must be an object"
+      )
+    );
+}
+
+export function validateRegisterObjectAPI(configurationObject) {
+  const validKeys = ["name", "load", "isActive", "customProps"];
+
+  const invalidPassedKeys = Object.keys(configurationObject).reduce(
+    (invalidKeys, prop) =>
+      validKeys.includes(prop) ? invalidKeys : invalidKeys.concat(prop),
+    []
+  );
+
+  if (invalidPassedKeys.length !== 0)
+    throw Error(
+      formatErrorMessage(
+        30,
+        __DEV__ &&
+          `The configuration object valid keys is: ${validKeys.join(
+            ", "
+          )}. Invalid keys: ${invalidPassedKeys.join(", ")}.`
+      )
+    );
+
+  validateArgumentsTypeAPI(
+    configurationObject.name,
+    configurationObject.load,
```

Comment:
Yeah, I'm no fan of the `load` prop either!
In my opinion, having both load/app would cause confusion, since it makes possible for wrongly defining both `load` and `app` in the same registerApplication. Having only 1 way to pass an App I think it's best!
`app` should be more friendly and obvious, since the term `Application` is being used in the documentation